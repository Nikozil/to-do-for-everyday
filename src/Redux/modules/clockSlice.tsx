import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { getTime } from 'date-fns';
import { AppThunk } from '../store';

export const initialState = {
  time: getTime(new Date()) as number,
  interval: undefined as number | undefined,
};

export type initialStateType = typeof initialState;
const clockSlice = createSlice({
  name: 'clock',
  initialState,
  reducers: {
    TICK: (state) => {
      return {
        ...state,
        time: getTime(new Date()),
      };
    },
    setClockInterval: (state, action: PayloadAction<any>) => {
      return {
        ...state,
        interval: action.payload,
      };
    },
  },
});

export const { TICK, setClockInterval } = clockSlice.actions;

export const startClock = (): AppThunk => (dispatch, getState) => {
  dispatch(stopClock());

  const interval = setInterval(() => {
    dispatch(TICK());
  }, 1000);

  dispatch(setClockInterval(interval));
};
export const stopClock = (): AppThunk => (dispatch, getState) => {
  clearInterval(getState().clock.interval);
  dispatch(setClockInterval(null));
};

export default clockSlice;
