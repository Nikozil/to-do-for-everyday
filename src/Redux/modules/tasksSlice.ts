import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import {
  add,
  endOfToday,
  format,
  getTime,
  milliseconds,
  startOfDay,
} from 'date-fns';

import { StoreAPI } from '../../api/StoreAPI';
import { AppThunk } from '../store';

export const initialState = {
  tasksList: [] as Task[],
  livedDay: {
    score: 0,
    tag: '',
    doneTasksList: [],
    timestamp: null,
  } as LivedDay,
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const tasksSlice = createSlice({
  name: 'tasks',
  initialState,
  reducers: {
    setTasks: (state, action: PayloadAction<Task[]>) => {
      return { ...state, tasksList: action.payload };
    },
    appendNewTask: (state, action: PayloadAction<Task>) => {
      state.tasksList.push(action.payload);
    },
    editTask: (state, action: PayloadAction<PartialTask>) => {
      const task = state.tasksList.find(
        (task) => task.id === action.payload.id
      );
      if (task) {
        task.data = { ...task.data, ...action.payload.data };
      }
    },
    removeTask: (state, action: PayloadAction<string>) => {
      state.tasksList = state.tasksList.filter(
        (task) => task.id !== action.payload
      );
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
    setLivedDay: (state, action: PayloadAction<LivedDay>) => {
      state.livedDay = { ...state.livedDay, ...action.payload };
    },
    appendTaskToDoneTasksList: (state, action: PayloadAction<LivedTask>) => {
      state.livedDay.doneTasksList.push(action.payload);
    },
    removeTaskToDoneTasksList: (state, action: PayloadAction<string>) => {
      const newdoneTasksList = state.livedDay.doneTasksList.filter(
        (task) => task.id !== action.payload
      );
      state.livedDay.doneTasksList = newdoneTasksList;
    },
    setTag: (state, action: PayloadAction<string>) => {
      state.livedDay.tag = action.payload;
    },
    setScore: (state, action: PayloadAction<Score>) => {
      state.livedDay.score = action.payload;
    },
    setTimestamp: (state, action: PayloadAction<number>) => {
      state.livedDay.timestamp = action.payload;
    },
  },
});

export const {
  setTasks,
  appendNewTask,
  editTask,
  removeTask,
  setInitStatus,
  setLivedDay,
  appendTaskToDoneTasksList,
  removeTaskToDoneTasksList,
  setTag,
  setScore,
  setTimestamp,
} = tasksSlice.actions;

export const getTasks = (): AppThunk => async (dispatch, getState) => {
  try {
    let tasks = await StoreAPI.getTask();
    let time = getState().clock.time;

    let livedDay = await StoreAPI.getLivedDay(format(time, 'dd.MM.yyyy'));

    if (!tasks) throw new Error('Не удалось получить данные');
    if (!livedDay) throw new Error('Не удалось получить данные');

    dispatch(setTasks(tasks));
    dispatch(setLivedDay(livedDay));
    dispatch(setInitStatus(true));
  } catch (err: any) {
    return err.message as string;
  }
};
export const addTask =
  (taskName: string, duration: Duration, repeat = 0): AppThunk =>
  async (dispatch, getState) => {
    const time = getTime(add(getState().clock.time, duration));
    const taskData = { name: taskName, done: false, time, repeat } as TaskData;
    try {
      const newTask = await StoreAPI.setTask(taskData);
      if (!newTask) throw new Error('Не удалось создать задачу');
      dispatch(appendNewTask(newTask));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const updateTask =
  (id: string, data: PartialTaskData): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.updateTask(id, data);
      dispatch(editTask({ id, data }));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const checkTask =
  (task: Task): AppThunk =>
  async (dispatch, getState) => {
    try {
      const { id, data } = task;
      const { name, done, time, repeat } = data;
      let newData;
      // make the data repeat date or complete
      if (repeat) {
        //move task for future
        let newTime = time;
        //count expiration of task + 1 repeat
        let expiration = Math.round(
          (getTime(endOfToday()) - newTime) /
            (milliseconds({ days: 1 }) * repeat) +
            1
        );
        //count date of future task
        newTime = getTime(add(newTime, { days: repeat * expiration }));
        newData = { time: newTime };
      } else {
        newData = { done: !done };
      }
      // add task to done list
      dispatch(checkupTimestamp());
      const doneTasks = {
        id: id,
        name: name,
      };
      //send data as batch
      const taskApiData = { taskId: id, taskData: newData };
      const doneTaskApiData = {
        doneTaskDate: format(getState().clock.time, 'dd.MM.yyyy'),
        doneTask: doneTasks,
        doneTaskRemove: false,
      };
      await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);
      //update store
      dispatch(editTask({ id, data: newData }));
      dispatch(appendTaskToDoneTasksList(doneTasks));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const uncheckTask =
  (task: LivedTask): AppThunk =>
  async (dispatch, getState) => {
    const { id, name } = task;
    try {
      let task = getState().tasks.tasksList.find((task) => task.id === id);
      if (task) {
        const { done, time, repeat } = task.data;
        let newData;
        // make the data repeat date or complete
        if (repeat) {
          //move date to past
          newData = { time: getTime(add(time, { days: -repeat })) };
        } else {
          //return task time and status
          newData = done
            ? { done: !done }
            : { done: false, time: getState().clock.time };
        }
        const taskApiData = { taskId: id, taskData: newData };
        const doneTask = {
          id: id,
          name: name,
        };
        const doneTaskApiData = {
          doneTaskDate: format(getState().clock.time, 'dd.MM.yyyy'),
          doneTask: doneTask,
          doneTaskRemove: true,
        };
        await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);

        //update store

        dispatch(editTask({ id, data: newData }));
        dispatch(removeTaskToDoneTasksList(id));
      } else throw new Error('Нет такой задачи');
    } catch (err: any) {
      return err.message as string;
    }
  };
export const deleteTask =
  (id: string): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.deleteTask(id);
      dispatch(removeTask(id));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const addTag =
  (tag: string): AppThunk =>
  async (dispatch, getState) => {
    try {
      dispatch(checkupTimestamp());
      const time = format(getState().clock.time, 'dd.MM.yyyy');
      const livedDay = { tag };
      await StoreAPI.updateLivedDay(time, livedDay);
      dispatch(setTag(tag));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const addScore =
  (score: Score): AppThunk =>
  async (dispatch, getState) => {
    try {
      dispatch(checkupTimestamp());
      const date = format(getState().clock.time, 'dd.MM.yyyy');
      const livedDay = { score };
      await StoreAPI.updateLivedDay(date, livedDay);
      dispatch(setScore(score));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const checkupTimestamp = (): AppThunk => async (dispatch, getState) => {
  if (!getState().tasks.livedDay.timestamp) {
    try {
      const date = format(getState().clock.time, 'dd.MM.yyyy');

      const timestamp = getTime(startOfDay(getState().clock.time));
      const livedDay = { timestamp };
      await StoreAPI.updateLivedDay(date, livedDay);
      dispatch(setTimestamp(timestamp));
    } catch (err: any) {
      return err.message as string;
    }
  }
};

export const doAgainTask =
  (task: Task): AppThunk =>
  async (dispatch, getState) => {
    try {
      const { id, data } = task;
      const { done } = data;

      //set date of tomorrow task
      const currentTime = getState().clock.time;
      const newTime = getTime(add(currentTime, { days: 1 }));

      const newData = { time: newTime, done: !done };

      await StoreAPI.updateTask(id, newData);
      //update store
      dispatch(editTask({ id, data: newData }));
    } catch (err: any) {
      return err.message as string;
    }
  };

export default tasksSlice;

export interface Task {
  id: string;
  data: TaskData;
}
export interface TaskData {
  name: string;
  done: boolean;
  time: number;
  repeat: number;
}
export type PartialTaskData = Partial<TaskData>;
export interface PartialTask {
  id: string;
  data: PartialTaskData;
}
export interface LivedTask {
  id: string;
  name: string;
}
export interface LivedDay {
  tag: string;
  score: Score;
  doneTasksList: LivedTask[];
  timestamp: null | number;
}
export type Score = 0 | 1 | 2 | 3 | 4 | 5;
