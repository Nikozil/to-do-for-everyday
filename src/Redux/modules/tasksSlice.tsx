import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { add, endOfToday, format, getTime, milliseconds } from 'date-fns';
import { StoreAPI } from '../../api/StoreAPI';
import { AppThunk } from '../store';

export const initialState = {
  tasksList: [] as Task[],
  doneTasksList: {} as DoneTask,
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const tasksSlice = createSlice({
  name: 'tasks',
  initialState,
  reducers: {
    setTasks: (state, action: PayloadAction<Task[]>) => {
      return { ...state, tasksList: action.payload };
    },
    editTask: (state, action: PayloadAction<PartialTask>) => {
      const task = state.tasksList.find(
        (task) => task.id === action.payload.id
      );
      if (task) {
        task.data = { ...task.data, ...action.payload.data };
      }
    },
    removeTask: (state, action: PayloadAction<string>) => {
      state.tasksList = state.tasksList.filter(
        (task) => task.id !== action.payload
      );
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
    setDoneTasks: (state, action: PayloadAction<DoneTask>) => {
      return { ...state, doneTasksList: action.payload };
    },
    addTaskToDoneTasksList: (state, action: PayloadAction<DoneTask>) => {
      return {
        ...state,
        doneTasksList: { ...state.doneTasksList, ...action.payload },
      };
    },
    removeTaskToDoneTasksList: (state, action: PayloadAction<string>) => {
      delete state.doneTasksList[action.payload];
    },
  },
});

export const {
  setTasks,
  editTask,
  removeTask,
  setInitStatus,
  setDoneTasks,
  addTaskToDoneTasksList,
  removeTaskToDoneTasksList,
} = tasksSlice.actions;

export const getTasks = (): AppThunk => async (dispatch) => {
  try {
    let tasks = await StoreAPI.getTask();
    let doneTasks = await StoreAPI.getDoneTask(
      format(new Date(), 'dd.MM.yyyy')
    );

    if (!tasks) throw new Error('Не удалось получить данные');
    if (!doneTasks) throw new Error('Не удалось получить данные');

    dispatch(setTasks(tasks));
    dispatch(setDoneTasks(doneTasks));
    dispatch(setInitStatus(true));
  } catch (err: any) {
    return err.message as string;
  }
};
export const addTask =
  (taskName: string, time: number, repeat = 0): AppThunk =>
  async (dispatch) => {
    const taskData = { name: taskName, done: false, time, repeat } as TaskData;
    try {
      await StoreAPI.setTask(taskData);
      dispatch(getTasks());
    } catch (err: any) {
      return err.message as string;
    }
  };
export const updateTask =
  (id: string, data: PartialTaskData): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.updateTask(id, data);
      dispatch(editTask({ id, data }));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const checkTask =
  (task: Task): AppThunk =>
  async (dispatch, getState) => {
    try {
      const { id, data } = task;
      const { name, done, time, repeat } = data;
      let newData;
      // make the data repeat date or complete
      if (repeat) {
        //move task for future
        let newTime = time;
        //count expiration of task + 1 repeat
        let expiration = Math.round(
          (getTime(endOfToday()) - newTime) /
            (milliseconds({ days: 1 }) * repeat) +
            1
        );
        //count date of future task
        newTime = getTime(add(newTime, { days: repeat * expiration }));
        newData = { time: newTime };
      } else {
        newData = { done: !done };
      }
      // add task to done list
      const doneTasksList = {
        [id]: name,
      };
      //send data as batch
      const taskApiData = { taskId: id, taskData: newData };
      const doneTaskApiData = {
        doneTaskDate: format(new Date(), 'dd.MM.yyyy'),
        doneTasks: doneTasksList,
        doneTasksMerge: true,
      };
      await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);
      //update store
      dispatch(editTask({ id, data: newData }));
      dispatch(addTaskToDoneTasksList(doneTasksList));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const uncheckTask =
  (id: string): AppThunk =>
  async (dispatch, getState) => {
    try {
      let task = getState().tasks.tasksList.find((task) => task.id === id);
      if (task) {
        const { done, time, repeat } = task.data;
        let newData;
        // make the data repeat date or complete
        if (repeat) {
          //move date to past
          newData = { time: getTime(add(time, { days: -repeat })) };
        } else {
          newData = { done: !done };
        }
        // add task to done list
        let doneTasksList = { ...getState().tasks.doneTasksList };
        delete doneTasksList[id];
        //send data as batch
        const taskApiData = { taskId: id, taskData: newData };
        const doneTaskApiData = {
          doneTaskDate: format(new Date(), 'dd.MM.yyyy'),
          doneTasks: doneTasksList,
          doneTasksMerge: false,
        };
        await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);

        //update store

        dispatch(editTask({ id, data: newData }));
        dispatch(removeTaskToDoneTasksList(id));
      } else throw new Error('Нет такой задачи');
    } catch (err: any) {
      return err.message as string;
    }
  };
export const deleteTask =
  (id: string): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.deleteTask(id);
      dispatch(removeTask(id));
    } catch (err: any) {
      return err.message as string;
    }
  };

export default tasksSlice;

export interface Task {
  id: string;
  data: TaskData;
}
export interface TaskData {
  name: string;
  done: boolean;
  time: number;
  repeat: number;
}
export type PartialTaskData = Partial<TaskData>;
interface PartialTask {
  id: string;
  data: PartialTaskData;
}
export interface DoneTask {
  [id: string]: string;
}
