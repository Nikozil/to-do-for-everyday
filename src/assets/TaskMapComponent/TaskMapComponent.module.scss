@import '../../App.scss';

.content_enter {
  opacity: 1;
  transition: opacity 400ms ease-in;
}
.content_enter-active {
  opacity: 0;
}
.content_exit {
  opacity: 1;
}
.content_exit-active {
  opacity: 0;
  transition: all 400ms ease-out;
}

.list {
  padding: 0;
}
.li {
  list-style-type: none;
  margin-bottom: 6px;
}

.taskList__comment {
  color: $comment-color;
  font-size: 0.8em;
}
