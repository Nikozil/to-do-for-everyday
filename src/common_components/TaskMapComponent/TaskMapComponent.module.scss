@import '../../App.scss';

.list {
  padding: 0;
  @media screen and (max-width: 767px) {
    font-size: 0.9em;
  }
}
.stub {
  color: $comment-color;
  font-size: 0.8em;
}

.content_enter {
  opacity: 1;
  transition: opacity 400ms ease-in;
}
.content_enter-active {
  opacity: 0;
}
.content_exit {
  opacity: 1;
}
.content_exit-active {
  opacity: 0;
  transition: all 400ms ease-out;
}

.content__li {
  list-style-type: none;
  margin-bottom: 6px;
}
