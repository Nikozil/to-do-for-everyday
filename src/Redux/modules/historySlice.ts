import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { endOfMonth, getTime, startOfMonth } from 'date-fns';

import { StoreAPI } from '../../api/StoreAPI';
import { AppThunk } from '../store';
import { LivedDay } from './tasksSlice';

export const initialState = {
  daysList: [] as HistoryDay[],
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const historySlice = createSlice({
  name: 'history',
  initialState,
  reducers: {
    setDays: (state, action: PayloadAction<HistoryDay[]>) => {
      return { ...state, daysList: action.payload };
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
  },
});

export const { setDays, setInitStatus } = historySlice.actions;

export const getDays =
  (month: number, year: number): AppThunk =>
  async (dispatch, getState) => {
    try {
      dispatch(setInitStatus(false));
      const date = getTime(new Date(year, month));
      const start = getTime(startOfMonth(date));
      const end = getTime(endOfMonth(date));

      const days = await StoreAPI.getDays(start, end);
      if (!days) throw new Error('Не удалось получить данные');

      dispatch(setDays(days));
      dispatch(setInitStatus(true));
    } catch (err: any) {
      return err.message as string;
    }
  };

export default historySlice;

export interface HistoryDay extends LivedDay {
  timestamp: number;
}
