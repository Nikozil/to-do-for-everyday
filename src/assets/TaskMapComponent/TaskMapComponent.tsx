import React, { useRef } from 'react';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import { LivedTask, Task } from '../../Redux/modules/tasksSlice';
import styles from './TaskMapComponent.module.scss';

const TaskMapComponent: React.FC<PropsType> = ({ list, stub, callback }) => {
  return (
    <>
      {list.length ? (
        <TransitionGroup component={'ul'} className={styles.list}>
          {list.map((task) => (
            <CSSTransitionWrapper
              task={task}
              callback={callback}
              key={task.id}
            />
          ))}
        </TransitionGroup>
      ) : (
        <span className={styles['taskList__comment']}>{stub}</span>
      )}
    </>
  );
};

export default TaskMapComponent;
const CSSTransitionWrapper: React.FC<WrapperPropsType> = ({
  task,
  callback,
  ...rest
}) => {
  const nodeRef = useRef(null);
  return (
    <CSSTransition
      {...rest}
      nodeRef={nodeRef}
      timeout={500}
      classNames={{
        enterActive: styles['content_enter-active'],
        enterDone: styles['content_enter'],
        exitActive: styles['content_exit-active'],
        exitDone: styles['content_exit'],
      }}>
      <li ref={nodeRef} className={styles.li}>
        {callback(task)}
      </li>
    </CSSTransition>
  );
};

interface PropsType {
  list: UnionTask[];
  stub: string;
  callback: (task: any) => JSX.Element;
}

interface WrapperPropsType {
  task: UnionTask;
  callback: (task: any) => JSX.Element;
}

type UnionTask = Task | LivedTask;
