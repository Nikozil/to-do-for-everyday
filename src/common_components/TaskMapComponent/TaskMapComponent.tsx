import React, { Dispatch, SetStateAction, useRef, useState } from 'react';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import { LivedTask, Task } from '../../Redux/modules/tasksSlice';
import styles from './TaskMapComponent.module.scss';

const TaskMapComponent: React.FC<PropsType> = ({ list, stub, callback }) => {
  const [currentOptions, setCurrentOptions] = useState<CurrentOptions>(null);

  return (
    <>
      {list.length ? (
        <TransitionGroup component={'ul'} className={styles.list}>
          {list.map((task) => (
            <CSSTransitionWrapper
              task={task}
              callback={callback}
              currentOptions={currentOptions}
              setCurrentOptions={setCurrentOptions}
              key={task.id}
            />
          ))}
        </TransitionGroup>
      ) : (
        <span className={styles.stub}>{stub}</span>
      )}
    </>
  );
};

export default TaskMapComponent;
const CSSTransitionWrapper: React.FC<WrapperPropsType> = ({
  task,
  currentOptions,
  setCurrentOptions,
  callback,
  ...rest
}) => {
  const nodeRef = useRef(null);
  return (
    <CSSTransition
      {...rest}
      nodeRef={nodeRef}
      timeout={500}
      classNames={{
        enterActive: styles['content_enter-active'],
        enterDone: styles['content_enter'],
        exitActive: styles['content_exit-active'],
        exitDone: styles['content_exit'],
      }}>
      <li ref={nodeRef} className={styles['content__li']}>
        {callback(task, currentOptions, setCurrentOptions)}
      </li>
    </CSSTransition>
  );
};

interface PropsType {
  list: UnionTask[];
  stub: string;
  callback: (
    task: any,
    currentOptions: CurrentOptions,
    setCurrentOptions: SetCurrentOptions
  ) => JSX.Element;
}

interface WrapperPropsType {
  task: UnionTask;
  currentOptions: CurrentOptions;
  setCurrentOptions: SetCurrentOptions;
  callback: (
    task: UnionTask,
    currentOptions: any,
    setCurrentOptions: any
  ) => JSX.Element;
}

type UnionTask = Task | LivedTask;
export type CurrentOptions = string | null;
export type SetCurrentOptions = Dispatch<SetStateAction<CurrentOptions>>;
