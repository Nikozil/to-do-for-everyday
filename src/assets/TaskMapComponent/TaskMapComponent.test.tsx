import { LivedTask } from '../../Redux/modules/tasksSlice';
import { render } from '../../utils/tests/test-utils';
import TaskMapComponent from './TaskMapComponent';

const list = [
  { id: '123', name: 'task1' },
  { id: '567', name: 'task2' },
] as LivedTask[];
const callback = (task: LivedTask) => <div>{task.name}</div>;

describe('Task Component tests', () => {
  it('rendering empty list', async () => {
    const { asFragment } = render(
      <TaskMapComponent list={[]} stub={'Empty'} callback={callback} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('rendering full list', async () => {
    const { asFragment } = render(
      <TaskMapComponent list={list} stub={'Empty'} callback={callback} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
