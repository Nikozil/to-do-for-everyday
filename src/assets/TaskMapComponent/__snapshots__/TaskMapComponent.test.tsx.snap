// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Component tests rendering empty list 1`] = `
<DocumentFragment>
  <span
    class="taskList__comment"
  >
    Empty
  </span>
   
</DocumentFragment>
`;

exports[`Task Component tests rendering full list 1`] = `
<DocumentFragment>
  <ul
    class="list"
  >
    <li
      class="li"
    >
      <div>
        task1
      </div>
    </li>
    <li
      class="li"
    >
      <div>
        task2
      </div>
    </li>
  </ul>
   
</DocumentFragment>
`;
